### PR TITLE
Use testing farm as GitHub action

### PR DESCRIPTION
**Pull request by phracek**
This pull request adds support for Testing Farm as GitHub Action.

It also renames `docker-tests.yml` -> `container-tests.yml`.

It also adds a fix for do not execute `build-and-push.yml` on forks.

### CODE REVIEW

**Comment by phracek** [comment]
[test]

**Comment by zmiklank** [comment]
@phracek, sorry, for noticing so late. 
Could you please adjust the PR same, as you adjusted https://github.com/sclorg/s2i-ruby-container/pull/348?
(And squash commits).

**Comment by phracek** [comment]
@zmiklank I have addressed all issues.

**Comment by zmiklank** [comment]
@phracek, thanks. 

Moreover, I think, that changes regarding build-and-push should not be in this PR or in TF commit (sorry, probably I wrote incorrectly the sentence about squashing commits - I meant to squash only commits, which should be squashed). But if you are OK with that, I think it is not a blocker. I would also consider to wait on the new build and push action to propagate, which already contains the condition you added (PR already exist https://github.com/sclorg/s2i-nodejs-container/pull/320).

**Comment by phracek** [comment]
[test-all]

**Comment by phracek** [comment]
[test-all]

**Comment by zmiklank** [comment]
Can we add here also this line?
https://github.com/sclorg/nginx-container/blob/c53969162182ac9cf6b3346e068efa7c73b6a23b/.github/workflows/container-tests.yml#L45

If not, I will create a separate PR.

**Comment by phracek** [comment]
[test-all]

File: .github/workflows/container-tests.yml [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+      - name: Schedule tests on external Testing Farm by Testing-Farm-as-github-action
+        id: github_action
+        uses: sclorg/testing-farm-as-github-action@v1
+        with:
+          api_key: ${{ secrets[matrix.api_key] }}
+          git_url: ${{ matrix.tmt_repo }}
+          git_ref: ${{ matrix.branch }}
+          tmt_plan_regex: ${{ matrix.tmt_plan }}
+          pull_request_status_name: ${{ matrix.context }}
+          variables: "REPO_URL=$GITHUB_SERVER_URL/$GITHUB_REPOSITORY;REPO_NAME=$GITHUB_REPOSITORY;PR_NUMBER=${{ steps.pr_nr.outputs.PR_NR }};OS=${{ matrix.os_test }};TEST_NAME=test"
```
Comment (zmiklank) [inline .github/workflows/container-tests.yml:72]:
```suggestion
          variables: "REPO_URL=$GITHUB_SERVER_URL/$GITHUB_REPOSITORY;REPO_NAME=$GITHUB_REPOSITORY;PR_NUMBER=${{ github.event.issue.number }};OS=${{ matrix.os_test }};TEST_NAME=test"
```

**Comment by phracek** [comment]
[test-all]

**Comment by phracek** [comment]
[test-openshift]

**Comment by phracek** [comment]
Fedora tests are failing for this reason:

`clients` pino:
```bash
make[2]: Leaving directory '/opt/app-root/src/node_modules/node-rdkafka/deps/librdkafka/src-cpp'
[91m/bin/sh: cmp: command not found
[0mmake -C examples
```

test `binary` test_run_binary_application:
```bash
npm ERR! command sh -c node-gyp rebuild
npm ERR! sh: /usr/lib/node_modules/npm/node_modules/@npmcli/run-script/lib/node-gyp-bin/node-gyp: Permission denied
[0m[91m
```

But these errors are expected sometimes.

File: .github/workflows/container-tests.yml (hunk 1 above)
Comment (zmiklank) [reply .github/workflows/container-tests.yml:72]: Sorry, I do not see this propagated, I think, that `steps.pr_nr.outputs.PR_NR` is invalid, as this step was removed.

**Review by zmiklank — APPROVED** [review]

**Pull request merged**